stockroom/storages/generic.py

```diff
@@ -0,0 +1,30 @@
+from .. import parser
+from ..repository import StockRepository
+
+
+# TODO: Maybe need another name than `GenericStore` but keeping it-
+# since scope of the name MetricStore seems to be too narrow
+
+class GenericStore:
+    """
+    A generic data storage class that does the functionality of
+    hangar metadata
+    """
+    def __init__(self, write):
+        self._write = write
+        self._repo = StockRepository()
+
+    def save(self, key, value):
+        co = self._repo.checkout(write=True)
+        co.metadata[parser.generic_metakey(key)] = value
+        co.close()
+
+    def load(self, key):
+        co = self._repo.checkout()
+        value = co.metadata[parser.generic_metakey(key)]
+        co.close()
+        return value
+
+
+def genericstore(write=False):
+    return GenericStore(write=write)
```

Comment:
Is this the constructor? Why not use the class directly?

Reply:
This is to keep the API uniform across different stores since datastore currently operates like a function that returns hangar checkout